src/hooks/useKeyboardNavigation.js

```diff
@@ -27,11 +29,68 @@ const useKeyboardNavigation = ({
 
   useEffect(() => {
     return tinykeys(emojiListRef.current, {
-      ArrowRight: focusNextEmoji,
-      ArrowLeft: focusPrevEmoji,
+      ArrowRight: () => navigateGrid(RIGHT),
+      ArrowLeft: () => navigateGrid(LEFT),
+      ArrowUp: () => navigateGrid(UP),
+      ArrowDown: () => navigateGrid(DOWN),
     });
   }, []);
 
+  const updateActiveItem = newActiveItem => {
+    if (newActiveItem && newActiveItem.firstChild)
+      newActiveItem.firstChild.focus();
+  };
+
+  const navigateGrid = direction => {
+    const grid = getCurrentEmojiListGroup();
+    const active = getActiveElement().parentElement;
+    const gridChildren = Array.from(grid.children);
+    const activeIndex = gridChildren.indexOf(active);
```

Comment:
~~This can be very costly since on every keystroke we would have to create an array of a couple thousand elements. Would it make sense to memoize it, or even just keep it as a value in a WeakMap?~~

Reply:
Or, better yet, we might be able to pull this off just by using indexOf. No need to convert it to an Array at all, just treat it as such.
```suggestion
    const activeIndex = Array.prototype.indexOf.call(grid.children, active)
```

Reply:
I'm not sure I understand.
![image](https://user-images.githubusercontent.com/7828909/151706122-5b8053f0-46d8-4e5e-ab50-94365f021dc4.png)

what should I do in the next lines that are using the gridChildren - 
should do also use Array.prototype ... call?

Reply:
For numOfItems and baseOffset you can use the element itself, for breakIndex you can Array.prototype..call